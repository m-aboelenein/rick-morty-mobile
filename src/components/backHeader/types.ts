import {ViewStyle} from 'react-native';

export interface BackHeaderStyles {
  headerContainer: ViewStyle;
  backButton: ViewStyle;
}

export interface BackHeaderProps {
  goBack: () => void;
}
