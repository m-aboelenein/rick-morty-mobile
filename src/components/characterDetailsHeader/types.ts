import {ImageStyle, TextStyle, ViewStyle} from 'react-native';
import {Character} from '../../hooks/useCharacter/types';

export interface CharacterDetailsHeaderProps {
  character: Character;
}

export interface CharacterDetailsHeaderStyles {
  container: ViewStyle;
  characterImage: ImageStyle;
  detailsContainer: ViewStyle;
  characterInfoContainer: ViewStyle;
  characterInfo: TextStyle;
}
