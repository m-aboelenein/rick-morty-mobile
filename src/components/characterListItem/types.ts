import {ImageStyle, TextStyle, ViewStyle} from 'react-native';
import {Character} from '../../hooks/useCharacters/types';

export interface CharacterListItemProps {
  character: Character;
  onPressCharacter: () => void;
}

export interface CharacterListItemStyles {
  container: ViewStyle;
  characterImage: ImageStyle;
  characterName: TextStyle;
}
