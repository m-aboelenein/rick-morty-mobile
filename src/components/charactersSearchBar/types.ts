import {ViewStyle} from 'react-native';

export interface CharactersSearchBarStyles {
  container: ViewStyle;
  searchField: ViewStyle;
}

export interface CharactersSearchBarProps {
  searchValue: string;
  setSearchValue: (value: string) => void;
}
