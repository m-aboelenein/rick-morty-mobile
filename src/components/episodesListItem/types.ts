import {TextStyle, ViewStyle} from 'react-native';
import {Episode} from '../../hooks/useCharacter/types';

export interface EpisodesListItemProps {
  episode: Episode;
}

export interface EpisodesListItemStyles {
  container: ViewStyle;
  airDate: TextStyle;
  name: TextStyle;
}
