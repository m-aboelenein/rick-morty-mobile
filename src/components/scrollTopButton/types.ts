import {ViewStyle} from 'react-native';

export interface ScrollTopButtonProps {
  onPressScroll: () => void;
}

export interface ScrollTopButtonStyles {
  container: ViewStyle;
}
