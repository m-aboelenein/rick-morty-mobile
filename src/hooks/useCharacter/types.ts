export interface CharacterDetailsQuery {
  character: Character;
}

export interface Episode {
  id: string;
  name: string;
  air_date: string;
}

export interface CharacterDetailsQueryVars {
  id: string;
}

export interface Character {
  id: number;
  name: string;
  species: string;
  gender: string;
  image: string;
  episode: Episode[];
}
