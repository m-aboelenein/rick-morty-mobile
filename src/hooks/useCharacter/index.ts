import {useLazyQuery} from '@apollo/client';
import {characterDetailsQuery} from '../../apollo/queries/characterDetails';
import {CharacterDetailsQuery, CharacterDetailsQueryVars} from './types';

const useCharacter = () => {
  const [getCharacterDetails, {data, loading, error, fetchMore}] = useLazyQuery<
    CharacterDetailsQuery,
    CharacterDetailsQueryVars
  >(characterDetailsQuery);
  return {
    data,
    loading,
    error,
    fetchMore,
    getCharacterDetails,
  };
};

export default useCharacter;
