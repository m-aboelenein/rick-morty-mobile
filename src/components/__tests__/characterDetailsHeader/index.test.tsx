import * as React from 'react';
import CharacterDetailsHeader from '../../characterDetailsHeader';
import renderer from 'react-test-renderer';
import {render} from '@testing-library/react-native';
import {Character} from '../../../hooks/useCharacter/types';

const testCharacter: Character = {
  id: 1,
  name: 'Rick Sanchez',
  gender: 'male',
  species: 'human',
  image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
  episode: [
    {
      id: '1',
      name: 'first episode',
      air_date: '1 jan 2021',
    },
  ],
};

it('renders a character header information', () => {
  const tree = renderer.create(
    <CharacterDetailsHeader character={testCharacter} />,
  );
  expect(tree).toMatchSnapshot();
});

it('renders the correct character data', () => {
  const {getByText} = render(
    <CharacterDetailsHeader character={testCharacter} />,
  );

  expect(getByText('Rick Sanchez')).not.toBe(null);
  expect(getByText('male')).not.toBe(null);
  expect(getByText('human')).not.toBe(null);

});
