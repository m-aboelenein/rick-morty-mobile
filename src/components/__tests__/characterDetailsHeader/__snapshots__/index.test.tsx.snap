// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a character header information 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 16,
      "flexDirection": "row",
      "marginHorizontal": 8,
      "minHeight": 120,
      "paddingHorizontal": 16,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 45,
          "height": 90,
          "marginRight": 16,
          "resizeMode": "cover",
          "width": 90,
        },
      ]
    }
  >
    <FastImageView
      resizeMode="cover"
      source={
        Object {
          "uri": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 16,
          }
        }
      >
        Name:
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "black",
            "fontSize": 16,
          }
        }
      >
        Rick Sanchez
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 16,
          }
        }
      >
        Species:
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "black",
            "fontSize": 16,
          }
        }
      >
        human
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 16,
          }
        }
      >
        Gender:
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "black",
            "fontSize": 16,
          }
        }
      >
        male
      </Text>
    </View>
  </View>
</View>
`;
